### PR TITLE
mpu: factorize stm32 mpu support

### PR DESCRIPTION
**Pull request by jamike**
- stm32 mpu configuration files have moved to a common directory to make possible the use for other
stm32 family.
- non regression test on nucleo_f401re : mpu_test and mpu_stack_guard test
-L4 support for all series is added since mpu is supported for all socs from the series (same as for f4)
- nucleo_l476rg and disco_l475_iot1 have passed mpu_test , mpu_stack_guard_test, and the mpu is enabled on these boards.
- F3 mpu support is added only for STM32F373XC (the other socs present in zephyr does support mpu)
- As I don't have board stm32373c_eval, I test the build but I did not push the commit that enables mpu on this board.

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
can you provide more comment in the subject of commit message for 'arm: soc: stm32: factorize mpu config'

**Review by galak — CHANGES_REQUESTED** [review]
Fix commit message for 'boards: nucleo: add MPU for nucleo_l476rg' missing signed-of-by

**Comment by erwango** [comment]
In commit messages:
please explain, for each family, if MPU is supported on whole family (should be yes) and then MPU is available or not depending on SoCs (not present on available F3, F1 SoCs as I understood).
Otherwise LGTM

File: arch/arm/soc/st_stm32/stm32f1/Kconfig.series [hunk 1]
```diff
@@ -12,6 +12,7 @@ config SOC_SERIES_STM32F1X
 	select SOC_FAMILY_STM32
 	select SYS_POWER_LOW_POWER_STATE_SUPPORTED
 	select HAS_STM32CUBE
+	select CPU_HAS_MPU
```
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f1/Kconfig.series]: Since MPU is not suported on all SoCs of stm32f1 series, this should be moved (if indeed supported) under Kconfig.defconfig.stm32f10...

File: arch/arm/soc/st_stm32/stm32f3/Kconfig.series [hunk 2]
```diff
@@ -13,6 +13,7 @@ config SOC_SERIES_STM32F3X
 	select SYS_POWER_LOW_POWER_STATE_SUPPORTED
 	select CPU_HAS_FPU
 	select CPU_HAS_SYSTICK
+	select CPU_HAS_MPU
```
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f3/Kconfig.series]: Since MPU is not suported on all SoCs of stm32f series, this should be moved (if indeed supported) under Kconfig.defconfig.stm32f3....

**Review by erwango — CHANGES_REQUESTED** [review]

File: arch/arm/soc/st_stm32/stm32f3/Kconfig.series (hunk 2 above)
Comment (jamike) [inline arch/arm/soc/st_stm32/stm32f3/Kconfig.series]: change performed

File: arch/arm/soc/st_stm32/stm32f1/Kconfig.series (hunk 1 above)
Comment (jamike) [inline arch/arm/soc/st_stm32/stm32f1/Kconfig.series]: Actually, I did not find a f1 having MPU support

**Comment by jamike** [comment]
I addressed all the above comments. Is something blocking the merge ?

File: arch/arm/soc/st_stm32/stm32f1/Kconfig.series (hunk 1 above)
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f1/Kconfig.series]: Hence, we consider CPU_HAS_MPU will be selected
- at series level if we consider all SoCs of a serie support MPU (F4 and L4 for now)
- at SoC level if, inside one series, some SoC support MPU, and some other don't
I'm alined. Though, it'd be nice to store this information somewhere (help section of config STM32_ARM_MPU_ENABLE config flag maybe?)

Otherwise LGTM

File: arch/arm/soc/st_stm32/common/Makefile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+obj-$(CONFIG_STM32_ARM_MPU_ENABLE) += arm_mpu_regions.o
+
+zephyr: $(KERNEL_HEX_NAME)
```
Comment (galak) [inline arch/arm/soc/st_stm32/common/Makefile]: Why is the hex image support moving here? If you want it common, do it as a separate patch/PR.

**Review by galak — CHANGES_REQUESTED** [review]
A few minor things:

1. rebase against latest arm branch
2. Remove the HEX image support from this PR.
3. Cleanup commit messages:
* body of commit messages should wrap around 70 char, you seem to break the lines sooner than need be
* remove spaces before ':' in subjects
* add space after ',' in commit body

**Comment by galak** [comment]
can you rebase on latest arm branch

**Comment by jamike** [comment]
rebase done on latest arm

**Review by galak — APPROVED** [review]
